sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -671,6 +674,57 @@ public String getId() {
         return withContext(context -> replaceItemInternal(itemType, itemId, doc, requestOptions, context));
     }
 
+    /**
+     * Run patch operations on an Item.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a single Cosmos item response with the replaced item.
+     *
+     * @param <T> the type parameter.
+     * @param itemId the item id.
+     * @param partitionKey the partition key.
+     * @param patchOperations patchOperations.
+     * @param itemType the item type.
+     * @return an {@link Mono} containing the Cosmos item resource response with the replaced item or an error.
+     */
+    public <T> Mono<CosmosItemResponse<T>> patchItem(
+        String itemId,
+        PartitionKey partitionKey,
+        List<PatchOperation<?>> patchOperations,
```

Comment:
any reason we are not taking similar route as TransactionalBatch where we hide the List of the operations behind a TransactionalBatch Object?

We could do something similar here too. We can define Patch class:
```java
Patch patch = Patch.createPatch(id, partitionKeyValue);
patch.addOperation(op1)
patch.addOperation(op2)
```

Any reason we are not following similar API?

this is for discussion only for now. I am "not" requesting to make any code change right now.

Reply:
Yeah. This is what is in Dotnet. But one reason I can think of is let's say user wants to do one increment operation continuously, then they can play around with that patch operation. But I will schedule a meeting and we can finalize it.